tensorflow/core/platform/posix/port.cc

```diff
@@ -47,6 +48,11 @@ string Hostname() {
   return string(hostname);
 }
 
+bool IsValidIpAddr(const string& ip) {
+  struct sockaddr_in sa;
+  return inet_pton(AF_INET, ip.c_str(), &(sa.sin_addr))!=0;
```

Comment:
`inet_pton()` only accepts textual IP addresses, so this will prevent users from providing hostnames in their `ClusterSpec`, and break backwards compatibility.